### PR TITLE
Add context menu to install a VSIX

### LINKED ISSUE #99459: Add install entry to vsix context menu
* have a vsix file in your workspace
* open its context menu
* there should be entry that enables you to install it

image

### PR DESCRIPTION
**Pull request by lszomoru**
This PR fixes #99459
* Moved installing the VSIX from the action to a command
* Command is being surfaced in the various places (context menu, more menu, dnd)

### CODE REVIEW

**Comment by isidorn** [comment]
Starting to review this. @sandy081 and @joaomoreno you do not have to, but can if you want to :)

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts [hunk 1]
```diff
@@ -257,6 +273,39 @@ CommandsRegistry.registerCommand({
 	}
 });
 
+CommandsRegistry.registerCommand({
+	id: EXTENSIONS_INSTALLVSIX_COMMAND_ID,
+	handler: async (accessor: ServicesAccessor, resources: URI[] | URI) => {
+		const extensions = Array.isArray(resources) ? resources : [resources];
```
Comment (isidorn) [inline src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:279]: Looks good, from what I see this is the same code as the Action code from before thus not going into too much detail.

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts [hunk 2]
```diff
@@ -65,6 +71,16 @@ Registry.as<IQuickAccessRegistry>(Extensions.Quickaccess).registerQuickAccessPro
 	helpEntries: [{ description: localize('manageExtensionsHelp', "Manage Extensions"), needsEditor: false }]
 });
 
+// Explorer
+MenuRegistry.appendMenuItem(MenuId.ExplorerContext, {
+	group: 'navigation',
```
Comment (isidorn) [inline src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:76]: Nice and clean registration.
I would just play with the order and group. This definetly should not be at the top of the navigation group as it is now - it is shown too high in the context menu.
Feels to me like it should be a seperate group, similar to Find in Folder

File: src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts [hunk 3]
```diff
@@ -60,6 +60,8 @@ import { IPreferencesService } from 'vs/workbench/services/preferences/common/pr
 import { DragAndDropObserver } from 'vs/workbench/browser/dnd';
 import { URI } from 'vs/base/common/uri';
 import { SIDE_BAR_DRAG_AND_DROP_BACKGROUND } from 'vs/workbench/common/theme';
+import { ICommandService } from 'vs/platform/commands/common/commands';
```
Comment (isidorn) [inline src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts:63]: Changes in this file look good.

File: src/vs/workbench/contrib/extensions/browser/extensionsActions.ts [hunk 4]
```diff
@@ -60,6 +60,7 @@ import { IProductService } from 'vs/platform/product/common/productService';
 import { IFileDialogService, IDialogService } from 'vs/platform/dialogs/common/dialogs';
 import { IProgressService, ProgressLocation } from 'vs/platform/progress/common/progress';
 import { Codicon } from 'vs/base/common/codicons';
+import { EXTENSIONS_INSTALLVSIX_COMMAND_ID } from 'vs/workbench/contrib/extensions/browser/extensions.contribution';
```
Comment (isidorn) [inline src/vs/workbench/contrib/extensions/browser/extensionsActions.ts:63]: Like a wrapper around a command, we do this also in explorer, I think it makes sense.

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
PR looks good! I would just like if you tackle the Order and Group of the action in the Explorer context menu.

**Comment by joaomoreno** [comment]
I yield my review! 💪

**Comment by bpasero** [comment]
@lszomoru  snow-plow alarm

**Comment by lszomoru** [comment]
@bpasero, I will make sure to get this in, this milestone.

**Comment by lszomoru** [comment]
Closing this PR in favor of https://github.com/microsoft/vscode/pull/108787

**Pull request closed without merging**